akka-http-jwt/src/test/scala/akka/http/jwt/scaladsl/server/directives/JwtDirectivesSpec.scala

```diff
@@ -0,0 +1,309 @@
+/*
+ * Copyright (C) 2024 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.http.jwt.scaladsl.server.directives
+
+import akka.http.jwt.internal.{ JwtClaimsImpl, JwtSprayJson }
+import akka.http.jwt.scaladsl.JwtSettings
+import akka.http.scaladsl.model.HttpResponse
+import akka.http.scaladsl.model.headers._
+import akka.http.scaladsl.server.AuthenticationFailedRejection.CredentialsRejected
+import akka.http.scaladsl.server.{ AuthenticationFailedRejection, AuthorizationFailedRejection, Directives, Route }
+import akka.http.scaladsl.testkit.ScalatestRouteTest
+import com.typesafe.config.ConfigFactory
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+import spray.json.{ JsArray, JsBoolean, JsNumber, JsObject, JsString, JsValue }
+
+import java.io.File
+import java.util.Base64
+
+class JwtDirectivesSpec extends AnyWordSpec with ScalatestRouteTest with JwtDirectives with Directives with Matchers {
+
+  def secret = "akka is great"
+
+  override def testConfigSource =
+    s"""
+       akka.loglevel = DEBUG
+       akka.http.jwt {
+         dev = off
+         realm = my-realm
+         secrets: [
+           {
+             key-id: my-key
+             issuer: my-issuer
+             algorithm: HS256
+             secret: "${Base64.getEncoder.encodeToString(secret.getBytes)}"
+           },
+           {
+             key-id: other-key
+             issuer: my-secondary-issuer
+             algorithm: HS256
+             secret: "${Base64.getEncoder.encodeToString("akka is better than great".getBytes)}"
+           }
+         ]
+       }
+      """
+
+  val basicClaims = Map[String, JsValue](
+    "sub" -> JsString("1234567890"),
+    "name" -> JsString("John Doe"),
+    "iat" -> JsNumber(1516239022))
+
+  def jwtHeader(claims: Map[String, JsValue] = basicClaims, secret: String = secret): Authorization = {
+    val token = JwtSprayJson.encode(JsObject("alg" -> JsString("HS256")), JsObject(claims), secret)
+    Authorization(OAuth2BearerToken(token))
+  }
+
+  def configTemplate(secret: String) = s"""
+          akka.loglevel = DEBUG
+          akka.http.jwt {
+            dev = off
```

Comment:
Would be good with one test case for the dev issuer as well

Reply:
Added unit tests for when `dev = on` .. I've removed the `dev-issuer` config since that is not used unless for signing (which we don't support).

https://github.com/akka/akka-http/pull/4377/commits/7d4fe8725a8ff77157f7defb6b5c9603d12362f5